### PR TITLE
feat: add deployment tests, remove unused scripts + GAE storage samples

### PR DESCRIPTION
**Pull request by bshaffer**

### CODE REVIEW

File: testing/run_test_suite.sh [hunk 1]
```diff
@@ -105,6 +105,12 @@ if ! type $TESTCMD > /dev/null; then
   exit 1
 fi
 
+if [ "${RUN_DEPLOYMENT_TESTS}" = "true" ]; then
+    TESTCMD="$TESTCMD --group deploy"
+else
+    TESTCMD="$TESTCMD --exclude-group deploy"
+fi
```
Comment (bshaffer) [inline testing/run_test_suite.sh:112]: New logic added so when `RUN_DEPLOYMENT_TESTS` is true, only the `deploy` group is run

**Review by ace-n — APPROVED** [review]

**Comment by ace-n** [comment]
@bshaffer taking another go at this. (Feel free to add fixes on top of this one!)

**Comment by ace-n** [comment]
[This PR](https://github.com/GoogleCloudPlatform/php-tools/pull/103) should fix [most of] the `functions` failures. Re-running tests now.

**Comment by ace-n** [comment]
Some of the functions tests took quite awhile.

@bshaffer I probably don't have time to continue working on this until next week, so would you mind reducing the retry limit on those? (If they haven't passed within ~7 minutes, they probably should fail outright.)

**Comment by ace-n** [comment]
**N.B:** @bshaffer I think the testing service account needs access to the Firestore-specific project (`php-docs-samples-kokoro-fs`).

**Comment by ace-n** [comment]
[This PR](https://github.com/GoogleCloudPlatform/php-tools/pull/108) was merged, so I'm rerunning the tests.

**Comment by ace-n** [comment]
@bshaffer looks like deployment tests are passing! 🎉

Would you mind fixing the unit tests though?

**Comment by bshaffer** [comment]
@ace-n the same tests have failed the last two nights in our nightly builds so I won't block this PR on them.

Nice work!!

**Pull request merged**